[docs] Update Expo Router installation guide (#26070)
expo/expo: docs/pages/router/installation.mdx

<Tab label="SDK 49 and above">

>     For the attibute 'name' use the `expo-router/entry` value in the **package.json** file. The initial client file is [**app/_layout.js**](/router/advanced/root-layout).

Reviewer: ```suggestion
    For the property `main`, use the `expo-router/entry` as its value in the **package.json**. The initial client file is [**app/_layout.js**](/router/advanced/root-layout).
```

The modification is incorrect. The property is `main` and not `name`. Also, for consistency across Expo docs, we call them properties as mentioned in the app config reference: https://docs.expo.dev/versions/latest/config/app/.
